akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -753,6 +766,82 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         errorCode should ===(ErrorCode.PROTOCOL_ERROR)
       }
     }
+
+    "delay stage completion" should {
+
+      "until in-flight responses are received" inAssertAllStagesStopped new TestSetup with NetProbes {
+        // Given an in-flight request...
+        user.emitRequest(HttpRequest(uri = "https://www.example.com/"))
+        network.expectDecodedResponseHEADERSPairs(1)
+        // ... and a completed user handler
+        user.requestOut.sendComplete()
+
+        network.sendFrame(
+          HeadersFrame(streamId = 1, endStream = true, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None)
+        )
+        // Then the user can still consume the response
+        user.expectResponse()
+      }
+
+      "until in-flight streams deliver the request and responses are received" inAssertAllStagesStopped new TestSetup(
+        Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 3)
+      ) with NetProbes {
+        // Given several in-flight requests...
+        val request = HttpRequest(uri = "https://www.example.com/")
+        user.emitRequest(request)
+        user.emitRequest(request)
+        user.emitRequest(request)
+        // ... and backpressured requests...
+        user.emitRequest(request) // this emit succeeds but is buffered
+
+        network.expect[HeadersFrame]().streamId shouldBe (1)
+        network.expect[HeadersFrame]().streamId shouldBe (3)
+        network.expect[HeadersFrame]().streamId shouldBe (5)
+        network.expectNoBytes(100.millis)
```

Comment:
Is this important to verify (wouldn't that belong in some other tests to verify nothing but the headers are emitted)?

Reply:
This ensures that the stream `7` is not just open, but not even seen on the network.

Then, this test ensures that both (1) streams expecting response, and (2) backpressured streams with a valid request ready but backpressured by MAX_CONCURRENT_STREAMS can complete.